tests/test_git.py

```diff
@@ -54,6 +59,22 @@ def test_read_file_from_private_repo(tmp_work_dir):
     assert output == b"# test-repository\nTesting GH permssions model\n"
 
 
+@pytest.mark.slow_test
+def test_commit_reachable_from_ref(tmp_work_dir):
+    should_be_reachable = commit_reachable_from_ref(
+        "https://github.com/opensafely/documentation",
+        "2170fadfeca7baf04f50376677253513e0a12bb1",
+        "1.6.0",
+    )
+    assert should_be_reachable
+    should_not_be_reachable = commit_reachable_from_ref(
+        "https://github.com/opensafely/documentation",
+        "afaee06237df52d78aa85920248572d5803379a3",
+        "1.6.0",
+    )
+    assert not should_not_be_reachable
```

Comment:
This is a bit of a brain teaser!

Naming suggestion (feel free to ignore 🙂) - call both variables `reachable` (or number them to distinguish), and then use the `assert`s to show the expectation on their outcome.

Reply:
Ha, yes! I obviously never vocalised this line in my head. Your suggestion is much better.